arangosh/Restore/RestoreFeature.cpp

```diff
@@ -1784,4 +1824,68 @@ Result RestoreFeature::getFirstError() const {
   return {TRI_ERROR_NO_ERROR};
 }
 
+void RestoreFeature::ProgressTracker::updateStauts(const std::string& filename,
+                                                   const RestoreFeature::CollectionStatus& status) {
+  std::unique_lock guard(_collectionStatesMutex);
```

Comment:
I would prefer if this method would just update the status under the lock, and then triggers an async writing of the file.
The async writing of the file can then write out the status of all collections without holding the mutex while doing file I/O.
Additionally, this could lead to less disk writes, as the async routine could write out multiple changes at once.

Reply:
Valid point.